frontend/lib/services/persistency_service.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:async';
+
+import 'package:frontend/models/event.dart';
+import 'package:frontend/models/sailing_club.dart';
+import 'package:frontend/models/boat.dart';
+
+// TODO: Should we have a percistency Interface that is segregated from the authentication interface?
+
+abstract class PersistencyService {
+//  final RegattaStore _store;
+
+//  PersistencyService(this._store);
+
+  Future signIn();
+  void signOut();
+
+  // These private methods are called by implementing classes
+  // to propagate changes in the DB to the frontend.
+/*
+  void newEvent(Event newEvent) {
```

Comment:
Hmm I am not sure about those methods. I think they only exist for the Firebase usecase. E.g. if we would have a databse or file as a backend, we would do the saving to disk and dispatching the success in a single method without the need of an async callback.